backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/pporder/api/impl/HUPPOrderBL.java

```diff
@@ -257,6 +280,28 @@ public List<HuId> retrieveAvailableToIssue(final I_PP_Order_BOMLine ppOrderBomLi
 				.collect(ImmutableList.toImmutableList());
 	}
 
+	@Override
+	public void issueAndReceiveMaturingHUs(@NonNull final I_PP_Order ppOrder)
+	{
+		final PPOrderId ppOrderId = PPOrderId.ofRepoId(ppOrder.getPP_Order_ID());
+		final I_M_HU huToBeIssued = getMaturingHU(ppOrder);
+
+		createIssueProducer(ppOrderId)
+				.considerIssueMethodForQtyToIssueCalculation(false) // issue exactly the HUs selected for maturing
+				.createIssues(ImmutableList.of(huToBeIssued));
+
+		receiveMaturedHU(ppOrder, huToBeIssued);
+
+		ppOrderBL.closeOrder(ppOrderId);
+	}
+
+	public boolean isAtLeastOneCandidateMaturing(@NonNull final List<I_PP_Order_Candidate> candidates)
+	{
+		return candidates
+				.stream()
+				.anyMatch(candidate -> candidate.isMaturing() && HuId.ofRepoIdOrNull(candidate.getIssue_HU_ID()) != null);
```

Comment:
refactor `candidate -> candidate.isMaturing() && HuId.ofRepoIdOrNull(candidate.getIssue_HU_ID()) != null` to a new method called something like isMaturingCandidate.

More, what's the reason of checking `HuId.ofRepoIdOrNull(candidate.getIssue_HU_ID()) != null`? Can't we do this check via an interceptor on PP_Order_Candidate, so we would enforce it from the very first start?
wdyt?